service/resource/deployment/resource.go

```diff
@@ -21,9 +21,14 @@ const (
 
 	createTimeout  = 5 * time.Minute
 	deploymentMode = "Incremental"
+	masterBranch   = "master"
 )
 
 type Config struct {
+	// URIVersion is used when creating template links for ARM templates.
+	// Defaults to master for deploying templates hosted on GitHub.
+	URIVersion string
+
```

Comment:
We usually structure configurations like this. 

* dependencies
* internals
* settings

Lets please keep this order and comment structure. Right now it is super easy to scan config, service, initialization and validation structures because we are almost consistent with this. See some random microservice: https://github.com/giantswarm/kubernetesd/blob/bea1a75406b6442b59ff04de51009b57462462d2/service/creator/service.go#L33.

Reply:
I think configuration shouldn't have `internals`. They should emerge in `New` based on the configuration IMO.

Reply:
I thought this is common sense. Thanks for proper clarification.

Reply:
Ack I missed that. The ordering is updated.